docs/webiny-apps/page-builder/development/creating-iframe-element-plugin.md

```diff
@@ -0,0 +1,182 @@
+
+
+When building pages, you are using these elements (e.g. Text, Image, Form, ...), which are registered in the system via plugins. In terms of rendering, every element has to have two plugins, one that defines how it's rendered in the actual page builder, and the other that defines how it's rendered on the website.
+
+If you are interested on different types of plugins, each app in Webiny consistes of the `types.ts` that includes all the plugin types of that app.
+In our case is [app-page-builder/src/types.ts](https://github.com/webiny/webiny-js/blob/master/packages/app-page-builder/src/types.ts)
+
+Let's render our plugin in the page builder, first we will use the `PbEditorPageElementPlugin` plugin type
+
+```jsx
+export default () => {
+    const PreviewBox = styled("div")({
+        textAlign: "center",
+        height: 50,
+        svg: {
+            height: 50,
+            width: 50
+        }
+    });
+
+    return [
+        {
+            name: "pb-editor-page-element-iframe",
+            type: "pb-editor-page-element",
+            elementType: "iframe",
+            toolbar: {
+                title: "iFrame",
+                group: "pb-editor-element-group-media",
+                preview() {
+                    return (
+                        <PreviewBox>
+                            <IFrameIcon />
+                        </PreviewBox>
+                    );
+                }
+            },
+            settings: ["pb-editor-page-element-settings-delete"],
+            onCreate: "open-settings",
+            create(options) {
+                return {
+                    type: "iframe",
+                    elements: [],
+                    data: {
+                        iframe: {
+                            url: "",
+                            height: 370
+                        },
+                        settings: {
+                            horizontalAlign: "center",
+                            margin: {
+                                desktop: { all: 0 },
+                                mobile: { all: 0 }
+                            },
+                            padding: {
+                                desktop: { all: 0 },
+                                mobile: { all: 0 }
+                            }
+                        }
+                    },
+                    ...options
+                };
+            },
+            render(props) {
+                return <IFrameEmbed {...props} />;
+            },
+            renderElementPreview({ width, height }) {
+                return <img style={{ width, height }} alt={"iFrame"} />;
+            }
+        } as PbEditorPageElementPlugin,
+    // Settings dialog comes here
+    ]
+```
+
+The key properties of the plugin are the create, data, settings, and render. They contribute to the state of our plugin. On changing the state, it’s up to the render function to render the component based on the state changes.
```

Comment:
![image](https://user-images.githubusercontent.com/5121148/81805844-df3a0a80-951b-11ea-95fa-0d48d60e7582.png)

Make the show as code, using backticks.